packages/core/src/index.ts

```diff
@@ -19,3 +22,8 @@ export * from './preview'
 export * from './provider'
 export * from './artifacts'
 export * from './types'
+export * from './errors'
+
+export const SphinxCore = {
+  compileAndExecuteDeployment,
+}
```

Comment:
I used this export pattern for `compileAndExecuteDeployment` so that I can stub it in `simulate.spec.ts` when testing `simulateDeploymentSubtask`. We can't pass a mocked function as an input parameter to `simulateDeploymentSubtask` (like [this](https://github.com/sphinx-labs/sphinx/blob/main/packages/core/src/networks.ts#L150-L153)) because this function is called as a Hardhat subtask.

If we don't export `compileAndExecuteDeployment` as a member of an object, this error occurs when attempting to stub it: `TypeError: Descriptor for property compileAndExecuteDeployment is non-configurable and non-writable`